### PR TITLE
Added ABI explanation to transactions page

### PR DESCRIPTION
**Pull request by qbzzt**
## Description

Add to the [transactions page](https://ethereum.org/en/developers/docs/transactions/#top) a description of what calldata means (in most cases)

### CODE REVIEW

File: src/content/developers/docs/transactions/index.md [hunk 1]
```diff
@@ -104,6 +104,35 @@ Example response:
 
 With the signature hash, the transaction can be cryptographically proven that it came from the sender and submitted to the network.
 
+### The data field {#the-data-field}
+
+The vast majority of transactions access a contract from an externally owned account (EOA).
```
Comment (minimalsm) [inline src/content/developers/docs/transactions/index.md:109]:
```suggestion
The vast majority of transactions access a contract from an externally-owned account.
```

File: src/content/developers/docs/transactions/index.md [hunk 2]
```diff
@@ -104,6 +104,35 @@ Example response:
...
+Most contracts are written in Solidity and interpret their data field in accordance with the [application binary interface (ABI)](https://docs.soliditylang.org/en/v0.8.12/abi-spec.html).
```
Comment (minimalsm) [inline src/content/developers/docs/transactions/index.md:110]:
```suggestion
Most contracts are written in Solidity and interpret their data field in accordance with the [application binary interface (ABI)](/glossary/#abi/).
```

File: src/content/developers/docs/transactions/index.md [hunk 3]
```diff
@@ -104,6 +104,35 @@ Example response:
...
+
+The first four bytes specify which function to call, using the hash of the function's name and arguments.
+You can sometimes identify the function from the selector using [this database](https://www.4byte.directory/signatures/).
+
+The rest of the calldata is the arguments, [encoded as specified in the ABI specs](https://docs.soliditylang.org/en/v0.8.12/abi-spec.html#formal-specification-of-the-encoding).
+
+For example, lets look at [this transaction](https://etherscan.io/tx/0xd0dcbe007569fcfa1902dae0ab8b4e078efe42e231786312289b1eee5590f6a1).
+Use **Click to see More** to see the calldata.
```
Comment (minimalsm) [inline src/content/developers/docs/transactions/index.md:118]: Wow etherscan capitalising the word more here is giving me bad feels 😅

File: src/content/developers/docs/transactions/index.md [hunk 4]
```diff
@@ -104,6 +104,35 @@ Example response:
...
+In this case [the contract source code](https://etherscan.io/address/0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48#code) has been uploaded to Etherscan, so we know the function is `transfer(address,uint256)`.
+
+The rest of the data is:
+
+```
+0000000000000000000000004f6742badb049791cd9a37ea913f2bac38d01279
+000000000000000000000000000000000000000000000000000000003b0559f4
+```
+
+According to the specifications integer nunbers, such as addresses, appear in the ABI as 32 byte words.
```
Comment (minimalsm) [inline src/content/developers/docs/transactions/index.md:130]: This sentence doesn't make sense @qbzzt, I'm not sure what it is supposed to say?

File: src/content/developers/docs/transactions/index.md [hunk 5]
```diff
@@ -104,6 +104,35 @@ Example response:
...
 
+### The data field {#the-data-field}
+
+The vast majority of transactions access a contract from an externally owned account (EOA).
+Most contracts are written in Solidity and interpret their data field in accordance with the [application binary interface (ABI)](https://docs.soliditylang.org/en/v0.8.12/abi-spec.html).
+
+The first four bytes specify which function to call, using the hash of the function's name and arguments.
+You can sometimes identify the function from the selector using [this database](https://www.4byte.directory/signatures/).
+
+The rest of the calldata is the arguments, [encoded as specified in the ABI specs](https://docs.soliditylang.org/en/v0.8.12/abi-spec.html#formal-specification-of-the-encoding).
```
Comment (minimalsm) [inline src/content/developers/docs/transactions/index.md:115]: Is there a way to make this link timeless? It'll technically be out of date at every patch upgrade (especially a problem for non-English languages which aren't as frequently translated)

**Review by minimalsm — CHANGES_REQUESTED** [review]

File: src/content/developers/docs/transactions/index.md (hunk 5 above)
Comment (minimalsm) [inline src/content/developers/docs/transactions/index.md:115]: There is @qbzzt! If we're linking to Solidity docs using /latest/ is probably the best link to use in most cases.

```suggestion
The rest of the calldata is the arguments, [encoded as specified in the ABI specs](https://docs.soliditylang.org/en/latest/abi-spec.html#formal-specification-of-the-encoding).
```

File: src/content/developers/docs/transactions/index.md [hunk 6]
```diff
@@ -104,6 +104,35 @@ Example response:
...
+In this case [the contract source code](https://etherscan.io/address/0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48#code) has been uploaded to Etherscan, so we know the function is `transfer(address,uint256)`.
+
+The rest of the data is:
+
+```
+0000000000000000000000004f6742badb049791cd9a37ea913f2bac38d01279
+000000000000000000000000000000000000000000000000000000003b0559f4
+```
+
+According to the specifications integer nunbers, such as addresses, appear in the ABI as 32 byte words.
```
Comment (minimalsm) [inline src/content/developers/docs/transactions/index.md:130]:
```suggestion
According to the specifications,  integer numbers (such as addresses) appear in the ABI as 32-byte words.
```

**Review by minimalsm — APPROVED** [review]

File: src/content/developers/docs/transactions/index.md (hunk 3 above)
Comment (qbzzt) [reply src/content/developers/docs/transactions/index.md:118]: Me too, but they can claim it's title case.

**Pull request merged**